app/Platform/Controllers/BeatenGamesLeaderboardController.php

```diff
@@ -90,136 +85,213 @@ public function __invoke(Request $request): View
-                'ImageIcon as GameIcon',
-                'ConsoleName',
-                DB::raw('MAX(AwardDate) as last_beaten_date'),
-            )
-            ->orderBy('rank_number')
-            ->groupBy('User');
+        $includedTypes = $this->getIncludedTypes($gameKindFilterOptions);
+        $typeBindings = $this->getSubQueryTypeBindings($includedTypes);
+
+        $systemIdCondition = $targetSystemId ? "AND r1.system_id = {$targetSystemId}" : "AND r1.system_id IS NULL";
+        $mostRecentGameIdSubquery = DB::raw("(
+            SELECT r1.game_id
+            FROM rankings AS r1
+            WHERE r1.user_id = rankings.user_id
+                {$systemIdCondition}
+                AND r1.type IN ({$typeBindings})
+            ORDER BY r1.updated_at DESC
+            LIMIT 1
+        ) AS most_recent_game_id");
+
+        $systemIdCondition = $targetSystemId ? "AND r2.system_id = {$targetSystemId}" : "AND r2.system_id IS NULL";
+        $lastBeatenDateSubquery = DB::raw("(
+            SELECT r2.updated_at
+            FROM rankings AS r2
+            WHERE r2.user_id = rankings.user_id
+                {$systemIdCondition}
+                AND r2.type IN ({$typeBindings})
+            ORDER BY r2.updated_at DESC
+            LIMIT 1
+        ) AS last_beaten_date");
+
+        $query = Ranking::select(
+            'user_id',
+            $mostRecentGameIdSubquery,
+            $lastBeatenDateSubquery,
+            DB::raw('RANK() OVER (ORDER BY SUM(value) DESC) as rank_number'),
+            DB::raw('ROW_NUMBER() OVER (ORDER BY SUM(value) DESC) as leaderboard_row_number'),
+            DB::raw('SUM(value) as total_awards'),
+        );
+
+        if ($targetSystemId) {
+            $query->where('system_id', $targetSystemId);
+        } else {
+            $query->whereNull('system_id');
+        }
+
+        if (!empty($includedTypes)) {
+            $query->whereIn('type', $includedTypes);
+        }
+
+        $query->groupBy('user_id')
+            ->orderBy('total_awards', 'desc')
+            ->orderBy('last_beaten_date', 'asc');
+
+        return $query;
     }
 
-    private function getLeaderboardRowCount(): int
+    private function determineGameKindFilterOptions(array $validatedData): array
     {
-        // SQLite, which is used for integration tests, doesn't support FOUND_ROWS().
-        // We'll naively return 25 for now.
-        if (DB::getDriverName() === 'sqlite') {
-            return 25;
+        $allFilterKeys = ['retail', 'hacks', 'homebrew', 'unlicensed', 'prototypes', 'demos'];
+
+        // As a safeguard, set everything to false by default.
+        $gameKindFilterOptions = array_fill_keys($allFilterKeys, false);
+
+        // Show retail games only by default. It will be the default filter choice.
+        $selectedKind = $validatedData['filter']['kind'] ?? 'retail';
+
+        switch ($selectedKind) {
+            case 'retail':
+                $gameKindFilterOptions['retail'] = true;
```

Comment:
Shouldn't retail also contain unlicensed? They were still sold at the store.

Reply:
You're right. I hadn't considered that. Updated in latest.